File: Java-Core/src/main/java/presto/javascript/JavaScriptLiteral.java
package presto.javascript;

import presto.utils.CodeWriter;

public abstract class JavaScriptLiteral implements JavaScriptExpression {

	String text;
	
	protected JavaScriptLiteral(String text) {
		this.text = text;
	}
	
	@Override
	public void toDialect(CodeWriter writer) {
		writer.append(text);
	}

}

File: Java-CodeServer/src/main/resources/js/prompto/javascript/JavaScriptLiteral.js
function JavaScriptLiteral(text) {
    this.text = text;
    this.value = eval(text);
    return this;
}

JavaScriptLiteral.prototype.interpret = function(context) {
    return this.value;
};

JavaScriptLiteral.prototype.toString = function() {
    return this.text;
};

JavaScriptLiteral.prototype.toDialect = function(writer) {
    writer.append(this.text);
};

exports.JavaScriptLiteral = JavaScriptLiteral;
